Sources/StreamChatUI/Appearance+Formatters/MessageDateSeparatorFormatter.swift

```diff
@@ -0,0 +1,28 @@
+//
+// Copyright © 2022 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+
+/// A formatter that converts the message date separator used in the message list to textual representation.
+/// This formatter is used to display the message date between each group of messages
+/// and the top date overlay in the message list.
+public protocol MessageDateSeparatorFormatter {
+    func format(_ date: Date) -> String
+}
+
+/// The default message date separator formatter.
+open class DefaultMessageDateSeparatorFormatter: MessageDateSeparatorFormatter {
+    public lazy var dateFormatter: DateFormatter = {
```

Comment:
We need to be careful with that. `lazy` vars are not thread safe.

Reply:
I'm seeing that this was already the case 🤔. We might need to review in the future lazy implementations that are accessed globally.

Reply:
That makes sense 👍 I can easily change this to `var`, will do it now 👍 DateFromatter's are lazily created either way only when a function is called (like `string(from:)`), so it is fine to have them has normal vars in this case.

Reply:
@polqf Done ✅